File: frontend/src/store/services/environment.ts
import { api } from './api'

export interface Environment {
  id: number
  name: string
  key: string
  description?: string
  base_url?: string
  is_active: boolean
  default_headers?: Record<string, any> | null
  default_params?: Record<string, any> | null
  variables?: Record<string, any> | null
  created_at?: string
  updated_at?: string
}

export interface EnvironmentCreate {
  name: string
  key: string
  description?: string
  base_url?: string
  is_active?: boolean
  default_headers?: Record<string, any> | null
  default_params?: Record<string, any> | null
  variables?: Record<string, any> | null
}

export interface EnvironmentUpdate extends Partial<EnvironmentCreate> {}

export const environmentService = {
  async getEnvironments(onlyActive = false): Promise<Environment[]> {
    const response = await api.get<Environment[]>('/environments', {
      params: onlyActive ? { only_active: true } : undefined,
    })
    return Array.isArray(response.data) ? response.data : []
  },

  async createEnvironment(data: EnvironmentCreate): Promise<Environment> {
    const response = await api.post<Environment>('/environments', data)
    return response.data
  },

  async updateEnvironment(id: number, data: EnvironmentUpdate): Promise<Environment> {
    const response = await api.put<Environment>(`/environments/${id}`, data)
    return response.data
  },

  async deleteEnvironment(id: number): Promise<void> {
    await api.delete(`/environments/${id}`)
  },
}

File: backend/app/schemas/environment.py
"""
环境相关的 Pydantic 模型
"""
from pydantic import BaseModel, Field, HttpUrl
from typing import Optional, Dict, Any
from datetime import datetime


class EnvironmentBase(BaseModel):
  """环境基础模型"""
  name: str = Field(..., min_length=1, max_length=100, description="环境名称")
  key: str = Field(..., min_length=1, max_length=50, description="环境标识，例如 dev/test/staging/prod")
  description: Optional[str] = Field(None, description="环境描述")
  base_url: Optional[str] = Field(None, description="基础URL，例如 https://api-test.example.com")
  is_active: bool = Field(default=True, description="是否启用")
  default_headers: Optional[Dict[str, Any]] = Field(
    default=None, description="默认请求头，例如 Authorization、X-Env 等"
  )
  default_params: Optional[Dict[str, Any]] = Field(
    default=None, description="默认查询参数"
  )
  variables: Optional[Dict[str, Any]] = Field(
    default=None, description="环境变量/业务变量，例如 tenantId、locale 等"
  )


class EnvironmentCreate(EnvironmentBase):
  """创建环境模型"""
  pass


class EnvironmentUpdate(BaseModel):
  """更新环境模型"""
  name: Optional[str] = Field(None, min_length=1, max_length=100)
  key: Optional[str] = Field(None, min_length=1, max_length=50)
  description: Optional[str] = None
  base_url: Optional[str] = None
  is_active: Optional[bool] = None
  default_headers: Optional[Dict[str, Any]] = None
  default_params: Optional[Dict[str, Any]] = None
  variables: Optional[Dict[str, Any]] = None


class EnvironmentResponse(EnvironmentBase):
  """环境响应模型"""
  id: int
  created_at: Optional[datetime] = None
  updated_at: Optional[datetime] = None

  class Config:
    from_attributes = True


